packages/components/src/unit-control/index.tsx

```diff
@@ -56,6 +57,15 @@ function UnforwardedUnitControl(
 	}: WordPressComponentProps< UnitControlProps, 'input', false >,
 	forwardedRef: ForwardedRef< any >
 ) {
+	if ( typeof unitProp !== 'undefined' ) {
```

Comment:
I think this is fine but in the interest of showing thorough consideration 🤓 I'll add we could look at doing it like:
```javascript
	if ( 'unit' in allProps ) {
```
Of course, that would rely on destructuring the props within the function body. The point is it's a little more strict and can warn more promptly. As an example, the Spacer block won't trigger the deprecation warning unless the user drags to resize.

Reply:
This is good point, I'll look into it (and likely implement it) once all dependant PRs are merged

Reply:
Done in https://github.com/WordPress/gutenberg/pull/39503/commits/d75a6be408fe0f5c9173784f28ece9bfc8b7e8a4